src/core/on-chain/index.ts

```diff
@@ -199,19 +205,40 @@ export const OnChainMixin = (superclass) =>
             sendAll,
           }
 
-          await lockExtendOrThrow({ lock, logger: onchainLoggerOnUs }, async () => {
-            const tx = await ledger.addOnUsPayment({
-              description: "",
-              sats,
-              metadata,
-              payerUser: this.user,
-              payeeUser,
-              memoPayer: memo,
-              shareMemoWithPayee: false,
-              lastPrice: UserWallet.lastPrice,
-            })
-            return tx
-          })
+          const price = await PriceService().getCurrentPrice()
+          if (price instanceof Error) throw price
+          const onChainFee = toSats(0)
+          const usd = sats * price
+          const usdFee = onChainFee * price
+
+          const payerWallet = await WalletsRepository().findById(this.user.id)
+          if (payerWallet instanceof CouldNotFindError) throw payerWallet
+          if (payerWallet instanceof Error) throw payerWallet
+          const recipientWallet = await WalletsRepository().findById(payeeUser.id)
+          if (recipientWallet instanceof CouldNotFindError) throw recipientWallet
+          if (recipientWallet instanceof Error) throw recipientWallet
```

Comment:
These are repeated lightning, I wonder if we should extract them into a new method.

Reply:
Potentially, it's done twice so far. I've been using a heuristic of not abstracting into methods until we repeat more than twice (guided by @bodymindarts on some "duplicated code" discussions we've had) or where the logic is potentially tricky (like with [this `checkAndVerifyTwoFA` helper method](https://github.com/GaloyMoney/galoy/pull/583/files#diff-8d34911386ecfd5c7a8de035b7d60a6981ac3911029b4e530b503f5009db7f70R35) I made)

Reply:
the two ifs are unnecessary, CouldNotFindError extends Error so you only need it once
```
if (payerWallet instanceof Error) throw payerWallet
```
and
```
if (recipientWallet instanceof Error) throw recipientWallet
```

Reply:
@dolcalmi yup but I wanted to leave it in until PR #588 in case we decide to add some separate logic to the `CouldNotFindError` branch instead of simply a return

Reply:
Example of separate logic [here](https://github.com/GaloyMoney/galoy/pull/602/files#diff-e43ce1074d48454f455720e70079bd3d8c6832daeb324523f6fddd5ba4cc7e81R14)